tools/sof-perf-analyzer.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2023 Intel Corporation. All rights reserved.
+
+'''
+The sof_trace_analyzer helps to analyze SOF trace output generated during the running of
+aplay and arecord commands or a test case in sof-test repository. It takes human readable
+trace file (contents from plain mtrace or the decoded dictionary trace) as input, and
+output below analysis results:
+    - Average and maximum MCPS of cpu average and cpu peak per SOF component
+
+Hardcoded value explained:
+DSP_CLK: The DSP clock frequency in Hz.
+DSP_TIMER: The DSP timer clock frequency in Hz.
+'''
+
+import re
+import pathlib
+import argparse
+from typing import Dict
+from typing import List
+from typing import TextIO
+from typing import Optional
+from typing import Generator
+from dataclasses import dataclass
+
+@dataclass()
+class TraceItem:
+    '''The structural representation for a single line of trace'''
+    timestamp: float
+    level: str
+    context: str # the trace context registered with LOG_MODULE_REGISTER
+    func: str
+    msg: str
+
+@dataclass(frozen=True)
+class Comp:
```

Comment:
Nit:
```suggestion
class Component:
```
Really not much longer and so much more readable.